exercises/variables/variables4.rs

```diff
@@ -1,10 +1,8 @@
 // variables4.rs
 // Execute `rustlings hint variables4` or use the `hint` watch subcommand for a hint.
 
-// I AM NOT DONE
-
 fn main() {
-    let x = 3;
+    let mut x = 3;
```

Comment:
I assume that `mut` enables ARC (Automatic Resource Counting) while let does a dealloc malloc operation on the same symbol.

I could see that `mut` is less desirable overall as its more costly to the GC.

Reply:
Its interesting because rust exposes the Reference Counter for smart pointers
https://doc.rust-lang.org/book/ch15-04-rc.html

Reply:
...is there GC though? I was under the impression that's the whole point of ownership and borrowing is you don't need GC; once the variable leaves scope, the memory is released without having to worry about it.

Reply:
The more you know... :)

So, what's the alternative here?